docs/adr/0001_cmsis_component_source_structure_recommendations.md

```diff
@@ -0,0 +1,117 @@
+# CMSIS component source structure recommendations
+Date: 2020-08-19
+
+## Status
+Proposed
+
+## Context
+
+[`ARM-software/CMSIS_5`](https://github.com/ARM-software/CMSIS_5) has been ported to Mbed OS. However, it is split across multiple directories and it is not immediately obvious that it comes from that directory. It also makes it difficult to create an independent CMake target can be built as a library that can be optionally be added by end users.
+
+The current proposal is to create a new directory `rtos/CMSIS_5/` that will mirror closely the structure used by `ARM-software/CMSIS_5/`. It will result in the removal of `rtos/source/TARGET_CORTEX/` and `cmsis/`.
+
+Additionally it will provide the following advantages:
+* Better maintenance of component
+* Easy creation of an independent CMake target
+* Easy source navigation that mirrors closely the upstream repository
+
+
+## Decision
+The following directory restructure is recommended:
+
+```
++-- rtos/CMSIS_5/
+|   +-- README.md
+|   +-- docs/
+|   +-- CMSIS/
+|       +-- include/
+|               +-- mbed_cmsis_conf.h   # Currently in cmsis/
+|       +-- RTE/
```

Comment:
Do we need a separate `RTE` directory?

Reply:
The creation of this directory was following the upstream repository.
The upstream repository has examples in it and the `RTE_Components.h` appears to be located within `RTE` directories containing other files in them. The `RTE` directory in this proposal was created in the event we have to add extra RTE related files.

See:
```
./CMSIS/RTOS2/RTX/Examples/TrustZoneV8M/RTOS_Faults/CM33_ns/RTE/_FVP_Simulation_Model/RTE_Components.h
./CMSIS/RTOS2/RTX/Examples/TrustZoneV8M/RTOS_Faults/CM33_s/RTE/_FVP_Simulation_Model/RTE_Components.h
./CMSIS/RTOS2/RTX/Examples/TrustZoneV8M/NoRTOS/CM33_ns/RTE/_FVP_Simulation_Model/RTE_Components.h
./CMSIS/RTOS2/RTX/Examples/TrustZoneV8M/NoRTOS/CM33_s/RTE/_FVP_Simulation_Model/RTE_Components.h
./CMSIS/RTOS2/RTX/Examples/TrustZoneV8M/RTOS/CM33_ns/RTE/_FVP_Simulation_Model/RTE_Components.h
./CMSIS/RTOS2/RTX/Examples/TrustZoneV8M/RTOS/CM33_s/RTE/_FVP_Simulation_Model/RTE_Components.h
./CMSIS/RTOS2/RTX/Examples/Blinky/RTE/_Simulation/RTE_Components.h
./CMSIS/RTOS2/RTX/Examples/MemPool/RTE/_Simulator/RTE_Components.h
./CMSIS/RTOS2/RTX/Examples/Migration/RTE/_Simulation/RTE_Components.h
./CMSIS/RTOS2/RTX/Examples/MsgQueue/RTE/_Simulator/RTE_Components.h
./CMSIS/NN/Examples/ARM/arm_nn_examples/gru/RTE/_ARMCM3/RTE_Components.h
./CMSIS/NN/Examples/ARM/arm_nn_examples/gru/RTE/_ARMCM4_FP/RTE_Components.h
./CMSIS/NN/Examples/ARM/arm_nn_examples/gru/RTE/_ARMCM0/RTE_Components.h
./CMSIS/NN/Examples/ARM/arm_nn_examples/gru/RTE/_ARMCM7_SP/RTE_Components.h
./CMSIS/NN/Examples/ARM/arm_nn_examples/cifar10/RTE/_ARMCM3/RTE_Components.h
./CMSIS/NN/Examples/ARM/arm_nn_examples/cifar10/RTE/_ARMCM4_FP/RTE_Components.h
./CMSIS/NN/Examples/ARM/arm_nn_examples/cifar10/RTE/_ARMCM0/RTE_Components.h
./CMSIS/NN/Examples/ARM/arm_nn_examples/cifar10/RTE/_ARMCM7_SP/RTE_Components.h
./CMSIS/NN/NN_Lib_Tests/nn_test/RTE/_ARMCM3/RTE_Components.h
./CMSIS/NN/NN_Lib_Tests/nn_test/RTE/_ARMCM4_FP/RTE_Components.h
./CMSIS/NN/NN_Lib_Tests/nn_test/RTE/_ARMCM0/RTE_Components.h
./CMSIS/NN/NN_Lib_Tests/nn_test/RTE/_ARMCM7_SP/RTE_Components.h
```

Reply:
Good.